File: dungeons-1.21.11/src/main/java/fishmod/features/item/ItemRarityHolder.java
package fishmod.features.item;

/**
 * Duck interface stamped onto {@code ItemStack} (via {@code ItemStackMixin}) to cache an item's
 * parsed rarity so the lore isn't re-scanned every frame for every visible slot. Ported from
 * blade-addons. Lives outside the mixin package so Mixin never pulls it into init.
 */
public interface ItemRarityHolder {
    ItemRarity fishmod$getItemRarity();
    boolean fishmod$hasItemRarity();
    void fishmod$setItemRarity(ItemRarity itemRarity);
    boolean fishmod$hasScanned();
}

File: dungeons-1.21.11/src/main/kotlin/fishmod/features/item/ItemRarityHolder.kt
package fishmod.features.item

/** Duck interface stamped onto `ItemStack` (via `ItemStackMixin`) to cache an item's parsed rarity so the lore isn't re-scanned every frame for every visible slot. */
interface ItemRarityHolder {
    fun `fishmod$getItemRarity`(): ItemRarity
    fun `fishmod$hasItemRarity`(): Boolean
    fun `fishmod$setItemRarity`(itemRarity: ItemRarity)
    fun `fishmod$hasScanned`(): Boolean
}
